src/v/kafka/server/handlers/configs/config_response_utils.h

```diff
@@ -0,0 +1,771 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/metadata_cache.h"
+#include "config/configuration.h"
+#include "config/data_directory_path.h"
+#include "config/node_config.h"
+#include "kafka/protocol/describe_configs.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/server/handlers/configs/config_response_utils.h"
+#include "kafka/server/handlers/topics/topic_utils.h"
+#include "kafka/server/handlers/topics/types.h"
+#include "kafka/server/request_context.h"
+#include "kafka/server/response.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+#include "model/namespace.h"
+#include "model/validation.h"
+#include "pandaproxy/schema_registry/subject_name_strategy.h"
+#include "reflection/type_traits.h"
+#include "security/acl.h"
+#include "ssx/sformat.h"
+#include "tristate.h"
+#include "utils/type_traits.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <fmt/ranges.h>
+
+#include <charconv>
+#include <optional>
+#include <string_view>
+#include <type_traits>
+
+namespace kafka {
+
+static inline bool config_property_requested(
+  const std::optional<std::vector<ss::sstring>>& configuration_keys,
+  const std::string_view property_name) {
+    return !configuration_keys.has_value()
+           || std::find(
+                configuration_keys->begin(),
+                configuration_keys->end(),
+                property_name)
+                != configuration_keys->end();
+}
+
+template<typename T>
+static inline void add_config(
```

Comment:
remove `static` from all the functions in this header, and also `inline` from any that are template functions. non-template free functions should have inline.

Reply:
Since I've moved the code to a .cc file instead of the header, I left the `inline`-ness of the code as it was previous to my PR, so I think this comment is outdated now.